File: projects/m2/014-sublists/solutions/giannighidoni/solution-sublists.js
const list = [1, 2, 3, 4, 5];
const subList = [1, 2, 5];

function isSublist(list, subList){
    if(subList.length === 0){
        return true;
    } else {  
            for(let iLis = 0; iLis < list.length; iLis++){
                if(subList[0] === list[iLis]){
                    if(subList.length === 1){
                        return true;
                    } else {
                        iLis++;
                        let count = 1;
                        for(let a = 1; a < subList.length; a++, iLis++){
                            if(subList[a] === list[iLis]){
                                count++;
                            }
                        }
                        if( count === subList.length){
                            return true;
                        } else {
                            return false;
                        }
                        break;  
                    }
                }
            }
    }
}

const res = isSublist(list, subList);

console.log(res);

File: projects/m2/014-sublists/solutions/jimmyhu/solution-sublists.py


def isSublist(lst_1,lst_2):
    if len(lst_2) == 0:
            return True
    elif len(lst_1) < len(lst_2):
        return False
    else:
        main_lst = lst_1
        sub_lst = lst_2  
    for element in sub_lst:
        if element not in main_lst:
            return False
    if len(sub_lst) > 1:
        for i in range(0,len(sub_lst)-1):
            if main_lst.index(sub_lst[i]) -1 != main_lst.index(sub_lst[i+1]) and main_lst.index(sub_lst[i]) +1 != main_lst.index(sub_lst[i+1]):
                return False
        return True
    else:
        return True       

def main():
    print('It is a sub list' if isSublist([1, 12, 3, 4, 11,15,17],[1, 12, 3, 4, 11,15,17]) else 'It is not a sublist'),
    print('It is a sub list' if isSublist([1, 12, 3, 4, 11,15,17],[]) else 'It is not a sublist'),
    print('It is a sub list' if isSublist([1, 12, 3, 4, 11,15,17],[1,17]) else 'It is not a sublist'),
    print('It is a sub list' if isSublist([1, 12, 3, 4, 11,15,17],[3, 4, 11]) else 'It is not a sublist'),
    print('It is a sub list' if isSublist([1, 12, 3, 4, 11,15,17],[1]) else 'It is not a sublist'),
    print('It is a sub list' if isSublist([1, 12, 3, 4, 11,15,17],[127]) else 'It is not a sublist'),
    print('It is a sub list' if isSublist([1, 12, 3, 4, 11,15,17],[4]) else 'It is not a sublist'),
    print('It is a sub list' if isSublist([1, 12, 3, 4, 11,15,17],[1, 12, 3, 4, 11,15,17,22]) else 'It is not a sublist'),
if __name__ == '__main__':
    main()
